### ISSUE TITLE
[Discovery] Dynatrace trial

### ISSUE THREAD

**Issue by wyattwalter** [department-of-veterans-affairs/va.gov-team#111]
@wyattwalter commented on [Fri May 17 2019](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/18623)

I volunteered us to participate in a trial of Dynatrace for vets-api (through Charles). We can get their agent installed at least on vets-api (maybe others, I'm not sure how it works yet) as a trial in at least dev and staging to see what kind of value it has.

Adding this to start tracking the work, though we are waiting on approval for firewall changes for their on-site deployment of Dynatrace.

**Comment by wyattwalter** [comment]
Meeting setup for tomorrow to kick this off 🎉

**Comment by wyattwalter** [comment]
Did our initial install today of some agents. The agent install needs to be automated yet. It can't go in the build of the images yet since those go to all environments, I'm going to stick them into the deployment config to be added to user-data for now.

Next steps:
- [ ] fixup haproxy stats permissions so the agent can pick that up
- [ ] waiting on a meeting about possible ruby APM support
- [ ] get AWS permissions updated in terraform to keep that consistent with what we had to change today
- [ ] get other users added (they just added me during our call since I was the one doing PoC installs)

**Comment by wyattwalter** [comment]
HAProxy support requires us to open up permissions a bit for the agent to read and write to the stats sock file. Article [here](https://www.dynatrace.com/support/help/technology-support/application-software/other-technologies/supported-out-of-the-box/haproxy/#socket-mode-monitoring). During this PoC, we're installing the agent at deploy time, and the agent installer they include adds the dtrace user. I'm not sure if we'll have issues with HAProxy not starting if the dtuser is defined in the config (probably will). This component might need a bit of work to make it work properly.

**Comment by b00klegger** [comment]
Agent is running across several applications in `dev`: revproxy, forwardproxy & vets-api.

We are at a point where the folks from Dynatrace want to begin instrumenting vets-api to continue the demo.

**Comment by wyattwalter** [comment]
We did some scoping of what it would take to get vets-api using Dynatrace for APM and it's similar in scope to implementing something like an opentrace library. For the moment, we don't have time to build this custom tracing into the API, so this is being tabled.

I'm going to remove the agent installation bits from our dev environments and will close this ticket off as wontfix.
